ambry-clustermap/src/main/java/com.github.ambry.clustermap/CloudServiceDataNode.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import org.json.JSONObject;
+
+import static com.github.ambry.clustermap.ClusterMapSnapshotConstants.*;
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+
+
+/**
+ * This is a place for {@link CloudServiceReplica}s to live on. This represents a location for in-process managed
+ * service clients (not a VCR like {@link CloudDataNode}). This will not contain any ports, as it is not meant to
+ * represent a remote network location. The hostname will just be the datacenter name. For logging purposes, the
+ * plaintext port will show as -1.
+ */
+class CloudServiceDataNode extends AmbryDataNode {
+
+  /**
+   * Instantiate a {@link CloudServiceDataNode}.
+   * @param dataCenterName the name of the dataCenter associated with this data node.
```

Comment:
Could this dataCenter be the same with regular datacenter (where regular ambry data nodes sit)? Or this should be a special datacenter representing the cloud?

Reply:
No, a cloud DC is currently modeled as a separate datacenter.